### PR TITLE
Remove release and cloud-ui testing docs

### PR DESCRIPTION
**Pull request by dchyun**
### :pushpin: Summary

The documentation on releases and testing in cloud ui have been moved to the following internal Confluence pages.

- [Release procedure](https://hashicorp.atlassian.net/wiki/x/HIBT0Q)
- [Testing in consumer's repositories](https://hashicorp.atlassian.net/wiki/x/24Fx0Q)

In this PR, these pages in the repo have been deleted, as the information is now captured in Confluence. 

### :link: External links

Jira ticket: [HDS-4242](https://hashicorp.atlassian.net/browse/HDS-4242)

[HDS-4242]: https://hashicorp.atlassian.net/browse/HDS-4242?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/RELEASE.md [hunk 1]
```diff
@@ -2,30 +2,49 @@
...
 
 In preparation for a minor or major release, login to Vercel CLI and create a canonical URL for the upcoming version:
 
-1. [Install Vercel CLI](https://vercel.com/docs/cli#installing-vercel-cli)
-2. Type `vercel login` in your terminal and select 'Continue with SAML Single Sign-On'
-3. Enter your team slug: `hashicorp` and press enter; you should be redirected to vercel.com with a 'CLI Login Success' message
-4. Go to the 'Version Packages' PR and identify the URL of the latest deployment (e.g. `hds-website-jeq5lwde8-hashicorp.vercel.app`)
-5. Determine the canonical URL for the version you're preparing to release (for version 4.3.0 it will be `hds-website-4-3-0.vercel.app`)
-6. Return to your terminal and create an alias using the vercel CLI `vercel alias hds-website-<deployment-id>-hashicorp.vercel.app hds-website-<version-number>.vercel.app` (e.g. `vercel alias hds-website-jeq5lwde8-hashicorp.vercel.app hds-website-4-3-0.vercel.app`)
+1. In GitHub, go to the “Version Packages” PR, which comes from the `changeset-release/main branch`
```
Comment (alex-ju) [inline packages/components/RELEASE.md:21]:
```suggestion
1. In GitHub, go to the “Version Packages” PR, which comes from the `changeset-release/main` branch
```

**Review by alex-ju — COMMENTED** [review]
I believe @Dhaulagiri's suggestion was for us to update this guidance and move both the Release (can be merged with your recent https://hashicorp.atlassian.net/wiki/x/HIBT0Q) and How-to-test guidance to Confluence altogether. I left a few comments, but as they move there we can remove them from here and update any reference to them.

**Comment by dchyun** [comment]
> I believe @Dhaulagiri's suggestion was for us to update this guidance and move both the Release (can be merged with your recent https://hashicorp.atlassian.net/wiki/x/HIBT0Q) and How-to-test guidance to Confluence altogether. I left a few comments, but as they move there we can remove them from here and update any reference to them.

If we want to full move these pages to confluence all on board with that to create one centralized place for release info. I can close this pr if that's the case. But if we want to keep them a little longer, can update them with my edits here.

**Comment by alex-ju** [comment]
> > I believe @Dhaulagiri's suggestion was for us to update this guidance and move both the Release (can be merged with your recent https://hashicorp.atlassian.net/wiki/x/HIBT0Q) and How-to-test guidance to Confluence altogether. I left a few comments, but as they move there we can remove them from here and update any reference to them.
> 
> If we want to full move these pages to confluence all on board with that to create one centralized place for release info. I can close this pr if that's the case. But if we want to keep them a little longer, can update them with my edits here.

It may be confusing to have them in both places. Would still be valuable to remove them in this PR after they get ported to Confluence.

**Review by alex-ju — APPROVED** [review]
Thank you!

**Pull request merged**